<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1600</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout" rowstretch="0,0,0,0">
    <property name="leftMargin">
     <number>5</number>
    </property>
    <property name="topMargin">
     <number>5</number>
    </property>
    <property name="rightMargin">
     <number>5</number>
    </property>
    <property name="bottomMargin">
     <number>5</number>
    </property>
    <property name="spacing">
     <number>5</number>
    </property>
    <item row="3" column="1" colspan="2">
     <widget class="QListWidget" name="listWidgetState">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>200</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <item>
       <property name="text">
        <string>상태: </string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>진행 시간: </string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>알고리즘 명: </string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>데이터 크기: </string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>속도 제한:</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>섞는 횟수: </string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>탐색 값: </string>
       </property>
      </item>
     </widget>
    </item>
    <item row="2" column="1" colspan="2">
     <widget class="QLabel" name="label_12">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>State</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="label_5">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Control</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="1" colspan="2">
     <widget class="QFrame" name="frameGraph">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>300</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayoutGraph"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="3" rowspan="3">
     <widget class="QListWidget" name="listWidgetLog">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>250</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>250</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="verticalScrollMode">
       <enum>QAbstractItemView::ScrollPerItem</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="1" colspan="2">
     <widget class="QLabel" name="label_11">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>Graph</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="0" rowspan="3">
     <widget class="QTabWidget" name="tabWidgetControl">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabSort">
       <attribute name="title">
        <string>정렬 알고리즘</string>
       </attribute>
       <widget class="QComboBox" name="comboBox_sort_select_Algorithm">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>10</y>
          <width>141</width>
          <height>22</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>선택 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>버블 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>삽입 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>병합 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>퀵 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>힙 정렬</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label_sort_select_algorithm">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>알고리즘 선택</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_sort_data_size">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>데이터 크기</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_sort_shuffle_number">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>100</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>섞는 횟수(회)</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_sort_speed_limit">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>속도 제한(s)</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="spinBox_sort_data_size">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>40</y>
          <width>141</width>
          <height>22</height>
         </rect>
        </property>
       </widget>
       <widget class="QSpinBox" name="spinBox_sort_shuffle_number">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>100</y>
          <width>141</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QSpinBox" name="spinBox_sort_speed_limit">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>70</y>
          <width>141</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QCheckBox" name="checkBox">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>100</y>
          <width>21</width>
          <height>21</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;섞는 모션 보이기&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QWidget" name="horizontalLayoutWidget">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>130</y>
          <width>271</width>
          <height>31</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="pushButton_sort_start">
           <property name="text">
            <string>실행</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_sort_restart">
           <property name="text">
            <string>다시하기</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_sort_stop">
           <property name="text">
            <string>중지</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QPushButton" name="testButton1">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>600</y>
          <width>91</width>
          <height>24</height>
         </rect>
        </property>
        <property name="text">
         <string>test button 1</string>
        </property>
       </widget>
       <widget class="QPushButton" name="testButton2">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>600</y>
          <width>91</width>
          <height>24</height>
         </rect>
        </property>
        <property name="text">
         <string>test button 2</string>
        </property>
       </widget>
       <widget class="QPushButton" name="testButton3">
        <property name="geometry">
         <rect>
          <x>200</x>
          <y>600</y>
          <width>91</width>
          <height>24</height>
         </rect>
        </property>
        <property name="text">
         <string>test button 3</string>
        </property>
       </widget>
       <widget class="QPushButton" name="testButton6">
        <property name="geometry">
         <rect>
          <x>200</x>
          <y>630</y>
          <width>91</width>
          <height>24</height>
         </rect>
        </property>
        <property name="text">
         <string>test button 6</string>
        </property>
       </widget>
       <widget class="QPushButton" name="testButton5">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>630</y>
          <width>91</width>
          <height>24</height>
         </rect>
        </property>
        <property name="text">
         <string>test button 5</string>
        </property>
       </widget>
       <widget class="QPushButton" name="testButton4">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>630</y>
          <width>91</width>
          <height>24</height>
         </rect>
        </property>
        <property name="text">
         <string>test button 4</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tabSearch">
       <attribute name="title">
        <string>탐색 알고리즘</string>
       </attribute>
       <widget class="QLabel" name="label_search_select_algorithm">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>알고리즘 선택</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_search_speed_limit">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>속도 제한(s)</string>
        </property>
       </widget>
       <widget class="QComboBox" name="comboBox_search_select_algorithm">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>10</y>
          <width>141</width>
          <height>22</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>선택 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>버블 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>삽입 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>병합 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>퀵 정렬</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>힙 정렬</string>
         </property>
        </item>
       </widget>
       <widget class="QSpinBox" name="spinBox_search_speed_limit">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>70</y>
          <width>141</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label_search_data_size">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>121</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>데이터 크기</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="spinBox_search_data_size">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>40</y>
          <width>141</width>
          <height>22</height>
         </rect>
        </property>
       </widget>
       <widget class="QLabel" name="label_search_search_number">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>100</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>탐색 값</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="spinBox_search_search_number">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>100</y>
          <width>141</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QCheckBox" name="checkBox_search_set_search_number">
        <property name="geometry">
         <rect>
          <x>60</x>
          <y>100</y>
          <width>21</width>
          <height>21</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;해제 시 랜덤 값&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QWidget" name="horizontalLayoutWidget_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>130</y>
          <width>271</width>
          <height>31</height>
         </rect>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="pushButton_search_start">
           <property name="text">
            <string>실행</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_search_restart">
           <property name="text">
            <string>다시하기</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_search_stop">
           <property name="text">
            <string>중지</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
    <item row="0" column="3">
     <widget class="QLabel" name="label_6">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>Log</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1600</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>파일</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>설정</string>
    </property>
    <addaction name="actionfps_set"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>도움말</string>
    </property>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>끝내기</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionfps_set">
   <property name="text">
    <string>fps set</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
